### ISSUE TITLE
KL divergence loss is summed while BCE is averaged 

### ISSUE THREAD

**Issue by Bidur-Khanal** [sinhasam/CRVAE#1]
While computing ELBO loss, I noticed that you have summed KLD across the batch and dimension, while BCE is average. This results in very high KLD loss and often leads to posterior collapse. I am not sure why have you chosen this strategy. Is there a reason to do so? Usually, both BCE and KLD should be same either both averaged or both summed.

```
def elbo_loss(mu, logvar, recon, original, beta):
    batch_size = original.size(0)
    BCE = F.binary_cross_entropy(
            recon.view(batch_size, -1), original.view(batch_size, -1), reduction='mean')

    # see Appendix B from VAE paper:
    # Kingma and Welling. Auto-Encoding Variational Bayes. ICLR, 2014
    # https://arxiv.org/abs/1312.6114
    # 0.5 * sum(1 + log(sigma^2) - mu^2 - sigma^2)
    KLD = -0.5 * torch.sum(1 + logvar - mu.pow(2) - logvar.exp())

    vae_loss = BCE + beta * KLD
    return vae_loss, BCE, KLD 


```
